File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/coupon/repository/IssuedCouponCustomRepositoryImpl.java
package band.gosrock.domain.domains.coupon.repository;

import static band.gosrock.domain.domains.coupon.domain.QIssuedCoupon.issuedCoupon;

import band.gosrock.domain.domains.coupon.domain.IssuedCoupon;
import com.querydsl.core.types.dsl.BooleanExpression;
import com.querydsl.jpa.impl.JPAQueryFactory;
import java.util.List;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class IssuedCouponCustomRepositoryImpl implements IssuedCouponCustomRepository {

    private final JPAQueryFactory queryFactory;

    @Override
    public List<IssuedCoupon> findAllByUserId(Long userId) {
        return queryFactory.selectFrom(issuedCoupon).where(userIdEq(userId)).fetch();
    }

    private BooleanExpression userIdEq(Long userId) {
        return userId == null ? null : issuedCoupon.userId.eq(userId);
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/coupon/repository/IssuedCouponCustomRepositoryImpl.kt
package band.gosrock.domain.domains.coupon.repository

import band.gosrock.domain.domains.coupon.domain.IssuedCoupon
import band.gosrock.domain.domains.coupon.domain.QIssuedCoupon.issuedCoupon
import com.querydsl.core.types.dsl.BooleanExpression
import com.querydsl.jpa.impl.JPAQueryFactory

class IssuedCouponCustomRepositoryImpl(
    private val queryFactory: JPAQueryFactory
) : IssuedCouponCustomRepository {

    override fun findAllByUserId(userId: Long): List<IssuedCoupon> =
        queryFactory.selectFrom(issuedCoupon).where(userIdEq(userId)).fetch()

    private fun userIdEq(userId: Long?): BooleanExpression? =
        if (userId == null) null else issuedCoupon.userId.eq(userId)
}
